From f805c9b368069bc96cb69edb6384d232c1fb3a6c Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Wed, 31 Jul 2019 16:29:14 -0400
Subject: cygwin: Use cygwin native timeval

__ms_timeval does not exist unless winsock.h or winsock2.h is included.
Which is not normally done when using the native GCC compiler.
---
 include/wx/private/socket.h | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/include/wx/private/socket.h b/include/wx/private/socket.h
index 56a3cef528..bcd5774a37 100644
--- a/include/wx/private/socket.h
+++ b/include/wx/private/socket.h
@@ -62,14 +62,7 @@
     #include <sys/time.h>   // for timeval
 #endif
 
-// 64 bit Cygwin can't use the standard struct timeval because it has long
-// fields, which are supposed to be 32 bits in Win64 API, but long is 64 bits
-// in 64 bit Cygwin, so we need to use its special __ms_timeval instead.
-#if defined(__CYGWIN__) && defined(__LP64__)
-    typedef __ms_timeval wxTimeVal_t;
-#else
-    typedef timeval wxTimeVal_t;
-#endif
+typedef timeval wxTimeVal_t;
 
 // these definitions are for MSW when we don't use configure, otherwise these
 // symbols are defined by configure
-- 
2.22.0.windows.1

